libraries/core_libs/consensus/src/pbft/pbft_manager.cpp

```diff
@@ -119,12 +119,9 @@ void PbftManager::run() {
     }
     // We need this section because votes need to be verified for reward distribution
     for (const auto &v : period_data.previous_block_cert_votes) {
-      vote_mgr_->addRewardVote(v);
-    }
-    if (!vote_mgr_->checkRewardVotes(period_data.pbft_blk)) {
-      LOG(log_er_) << "Invalid reward votes in block " << period_data.pbft_blk->getBlockHash() << " in DB.";
-      assert(false);
+      vote_mgr_->verifyRewardVote(v);
```

Comment:
Shouldn't we add vote after this verification?

Reply:
no, the reason for verification is that verification sets some internal structure within the votes, vote manager holds reward votes for the pbft block that is not yet part of the chain and this is not the case, here pbft block is already pushed to chain just not finalized/executed